interfaces/apparmor/backend.go

```diff
@@ -446,6 +446,29 @@ func addUpdateNSProfile(snapInfo *snap.Info, opts interfaces.ConfinementOptions,
 	}
 }
 
+func downgradeConfinement() bool {
+	kver := osutil.KernelVersion()
+	switch {
+	case release.DistroLike("opensuse-tumbleweed"):
+		if cmp, _ := strutil.VersionCompare(kver, "4.16"); cmp >= 0 {
+			// As a special exception, for openSUSE Tumbleweed which ships Linux
+			// 4.16, do not downgrade the confinement template.
+			return false
+		}
+	case release.DistroLike("arch"):
+		if !strings.HasSuffix(kver, "-hardened") {
+			// how did we get here anyway?
```

Comment:
As of now, `linux-hardened` is the only kernel in the repositories that has AppArmor enabled. The comment was to note that getting here with a different kernel is unexpected. If 4.18.6 arch kernel is really released with AppArmor this will need to be updated accordingly